enhancements/dns/dns-operator-operand-logging-level.md

```diff
@@ -0,0 +1,257 @@
+the Operator and CoreDNS issues easier for cluster administrators and OpenShift developers.
+
+For logging purposes, CoreDNS defines several classes of responses, such as error, denial and all.
+* denial: either NXDOMAIN or nodata responses (Name exists, type does not). A nodata response sets the return code to NOERROR.
+* error: SERVFAIL, NOTIMP, REFUSED, etc. Anything that indicates the remote server is not willing to resolve the request.
+* all: all responses, including successful responses, errors, and denials.
+A logging level API for CoreDNS logs would assist cluster administrators who wish to have more control
+over CoreDNS logs.
+
+Also, a logging level API for the DNS Operator would assist OpenShift developers working on the DNS Operator who may
+desire more in-depth logging statements when working on the operator's controllers.
+
+* Some users want to add new prometheus alert 'CoreDNS is returning SERVFAIL for X% of requests alert' to the recent updates of OCP.
+  Adding this Prometheus alert is useful, but it would be more useful if we could see which requests were getting SERVFAIL responses.
+  So we would like to configure the log plugin for CoreDNS to log queries.
+
+* Some user want to avoid use of tcpdump to see the queries and want log plugin to be enabled to log queries in coredns.
+
+### Goals
+
+Add a user-facing API for controlling the run-time verbosity of the [OpenShift DNS Operator and CoreDNS](https://github.com/openshift/cluster-dns-operator).
+### Non-Goals
+
+* Change the default logging verbosity of the DNS Operator or CoreDNS in production OCP clusters.
+
+## Proposal
+
+### DNS Operator Log Level API
+We will be defining a new API field `operatorLogLevel` in `DNSSpec` with newly defined type `DNSLogLevel`.
+This type is similar to the existing `LogLevel` type except that the values of `DNSLogLevel` are a subset of the values of `LogLevel`.
+Valid values will be the following: "Normal", "Debug", "Trace".
+We will use [logrus](https://github.com/sirupsen/logrus#level-logging) to set the log level for the operator-level logging.
+Logrus has seven logging levels: Trace, Debug, Info, Warning, Error, Fatal and Panic.
+```go
+log.Trace("Something very low level.")
+
+log.Debug("Useful debugging information.")
+
+log.Info("Something noteworthy happened!")
+
+log.Warn("You should probably take a look at this.")
+
+log.Error("Something failed but I'm not quitting.")
+
+// Calls os.Exit(1) after logging:
+log.Fatal("Bye.")
+
+// Calls panic() after logging:
+log.Panic("I'm bailing.")
+```
+After the logging level on a Logger is set, log entries with that severity or anything above it will be logged.
+For example, `log.SetLevel(log.InfoLevel)` will log anything that is info or above (warn, error, fatal, panic).  This is the default log level.  
+So, we will be reading `operatorLogLevel` in a separate controller to watch dnses and setting log level.
```

Comment:
You didn't end up using a separate controller, so this text should be updated when you get the chance.